json-core/src/main/java/io/micronaut/json/JsonSyntaxException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.json;
+
+import java.io.IOException;
+
+/**
+ * Exception thrown when there is a syntax error in JSON (e.g. mismatched braces).
+ */
+public final class JsonSyntaxException extends IOException {
```

Comment:
add `@since` and `author` tags. Add constructor that takes a message and the exception.

Reply:
I intentionally only allowed these two constructors, one for "delegating" to an implementation exception (e.g. JsonParseException), one for a message thrown by micronaut code (i.e. a syntax error detected by JsonCounter). The behavior does not exactly match the super constructors: The JsonSyntaxException(Throwable) constructor copies the message to the JsonSyntaxException, so that the JsonError that is shown to the user contains the proper message.